packages/node/src/exec.ts

```diff
@@ -1,26 +1,63 @@
 /// <reference types="../types/shellwords" />
 import { split } from 'shellwords';
 
-import { Task } from '@effection/core';
-import { ExecOptions, Process, } from './exec/api';
+import { Task, Operation } from '@effection/core';
+import { ExecOptions, Process, ProcessResult, CreateOSProcess } from './exec/api';
 import { createPosixProcess } from './exec/posix';
 import { createWin32Process, isWin32 } from './exec/win32';
 
 export * from './exec/api';
+export * from './exec/error';
+
+export interface Exec {
+  run(scope: Task): Process;
+  join(): Operation<ProcessResult>;
+  expect(): Operation<ProcessResult>;
+}
+
+const createProcess: CreateOSProcess = (scope, cmd, opts) => {
+  if (isWin32()) {
+    return createWin32Process(scope, cmd, opts);
+  } else {
+    return createPosixProcess(scope, cmd, opts);
+  }
+};
 
 /**
  * Execute `command` with `options`. You should use this operation for processes
  * that have a finite lifetime and on which you may wish to synchronize on the
  * exit status. If you want to start a process like a server that spins up and runs
  * forever, consider using `daemon()`
  */
-export function exec(scope: Task, command: string, options: ExecOptions = {}): Process {
+export function exec(command: string, options: ExecOptions = {}): Exec {
   let [cmd, ...args] = split(command);
   let opts = { ...options, arguments: args.concat(options.arguments || []) }
 
-  if (isWin32()) {
-    return createWin32Process(scope, cmd, opts);
-  } else {
-    return createPosixProcess(scope, cmd, opts);
+  return {
+    run(scope: Task) {
+      return createProcess(scope, cmd, opts);
+    },
```

Comment:
It feels more consistent to follow the established pattern here, even when we're delegating:

```ts
run(scope: Task) {
  return createProcess(cmd, opts).run(scope);
}
```

Reply:
I chose not to change this API because it is internal, and I wanted to keep the diff smaller, but I can totally change that if you prefer.